src/storage/transaction/TransactionManager.h

```diff
@@ -129,12 +128,18 @@ class TransactionManager {
   // this is a callback register to Part::onLostLeadership
   void onLeaderLostWrapper(const ::nebula::kvstore::Part::CallbackOptions& options);
 
+  void waitUntil(std::function<bool()>&& cond, folly::Promise<folly::Unit>&& p);
+
  protected:
   using SpacePart = std::pair<GraphSpaceID, PartitionID>;
 
   StorageEnv* env_{nullptr};
-  std::shared_ptr<folly::IOThreadPoolExecutor> exec_;
+  std::shared_ptr<folly::IOThreadPoolExecutor> worker_;
+  std::shared_ptr<folly::IOThreadPoolExecutor> controller_;
```

Comment:
Can you add comments to these two variables?

Reply:
ok